Backups: use DateRangePicker to filter DataViews

myhro opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of [DOTDASH-436][1].

## Proposed Changes

- Add a `DateRangePicker` component to the Backups list to allow users to filter backup entries by date
  range, replacing the previous DataViews date column filtering.
- Extract common date range logic from the Logs page into the `useDateRange` hook that handles URL synchronization, timezone-aware date calculations, and optional auto-refresh functionality.
- Add `after` and `before` parameters to the backup activity log API fetcher/query to support server-side date filtering.
-  Update backup selection logic to handle filtered results, updating selection and resetting to the first page (like Logs do) when date ranges change.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

We concluded that the default `DataViews` [date filter isn't ideal and we should switch][2] to `DateRangePicker`.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Go to `/v2/sites/{site}/backups`.
- Select different dates/presets from the calendar date picker right beside the `Back up now` button:

<img width="775" height="620" alt="image" src="https://github.com/user-attachments/assets/4c1e6a0e-a398-4cd3-ad66-e480398619fc" />

### Today

<img width="1256" height="295" alt="image" src="https://github.com/user-attachments/assets/469bea2e-8a95-402c-a0c4-b7a9393c87d3" />

### Yesterday

<img width="1251" height="287" alt="image" src="https://github.com/user-attachments/assets/48c245d2-0b48-4737-bbe9-2694fa6a9d42" />

### Last 30 days

<img width="1253" height="1203" alt="image" src="https://github.com/user-attachments/assets/058c52cb-1ed2-4925-8df7-840bc949de04" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

[1]: https://linear.app/a8c/issue/DOTDASH-436
[2]: https://linear.app/a8c/issue/DOTDASH-426/#comment-d0fd3443

myhro committed:
Backups: use DateRangePicker to filter DataViews

arthur791004 commented on packages/api-core/src/site-activity-log/fetchers.ts:
```diff
@@ -18,16 +18,36 @@ export async function fetchSiteActivityLog(
 		},
-		{
-			number,
-			aggregate,
```
The undefined properties will be filtered out by default so we can just add `before` and `after` here directly.

myhro replied:
Indeed, thanks for the suggestion. In the `Overview` page I'm seeing it being called with `{number: 1, aggregate: false, after: undefined, before: undefined}` and the query is just `/wpcom/v2/sites/:id/activity/rewindable?_envelope=1&number=1&aggregate=false`, without `before`/`after` parameters.

arthur791004 commented on client/dashboard/app/hooks/use-date-range.ts:
```diff
@@ -0,0 +1,80 @@
+	autoRefresh?: boolean;
+}
+
+export function useDateRange( {
```
I'm not sure, but I don't think we've decided to have global hooks under Dashboard V2 yet. Currently, site-performance is the only one here, and it will be moved to the Performance page once that work starts. cc @Automattic/lego Any thoughts on global hooks?

fushar replied:
I think it's fine. In this PR, this `useDateRange()` is being used in backups and logs, and it seems generic enough. Unless there's a better place to put it 🤔

myhro replied:
The goal is to share it between backups and logs indeed. Should we name it something like `__useDateRange` to suggest it shouldn't be used outside the Hosting Dashboard? I'm also open to suggestions about where to better place it.

Initsogar replied:
> The goal is to share it between backups and logs indeed. Should we name it something like __useDateRange to suggest it shouldn't be used outside the Hosting Dashboard?

I think `useDateRange` is fine. I don't expect hooks on Hosting Dashboard V2 to be used elsewhere.

I am also curious about best practices for where to locate hooks.

arthur791004 replied:
We talked about it before (p1747933245016909-slack-CRWCHQGUB) — I’m not sure of the answer, so yeah, I guess it’s fine to put it there.

arthur791004 commented on client/dashboard/utils/datetime.ts:
```diff
@@ -290,3 +298,88 @@ export function formatYmd( date: Date, timezoneString?: string, gmtOffset?: numb
+	if ( timezoneString ) {
+		const startYmd = dateI18n( 'Y-m-d', start, timezoneString );
+		const endYmd = dateI18n( 'Y-m-d', end, timezoneString );
+		const startSec = +dateI18n( 'U', `${ startYmd } 00:00:00`, timezoneString );
```
We should not use `+` to do the conversion.

fushar replied:
Sorry -- why? (Or what's the correct way)

myhro replied:
I'm [keeping this function in its current state][1] at `logs/utils.ts`, so we don't have to worry about it. I'm still curious about not using `+`, though.

[1]: https://github.com/Automattic/wp-calypso/blob/eb9eeba281ff8cdf3d7029085fcb680ab465704d/client/dashboard/sites/logs/utils.ts#L25

arthur791004 replied:
Oh okay. Sorry I didn't notice that...

Initsogar commented on client/dashboard/utils/datetime.ts:
```diff
@@ -290,3 +298,88 @@ export function formatYmd( date: Date, timezoneString?: string, gmtOffset?: numb
+/**
+ * Get the default date range for date-filtered pages (7 days ending today).
+ */
+export function getDefaultDateRange( timezoneString?: string, gmtOffset?: number ) {
```
It looks like `getDefaultDateRange` is specific to the useDateRange hook. Maybe we could just place it in the same file?

myhro replied:
Previously [it was coming][1] from `logs/utils.ts`, hence why I kept it separately. It makes sense to keep it together with the hook implementation, though. I'm updating it.

[1]: https://github.com/Automattic/wp-calypso/blob/eb9eeba281ff8cdf3d7029085fcb680ab465704d/client/dashboard/sites/logs/index.tsx#L35-L39

Initsogar commented on client/dashboard/utils/datetime.ts:
```diff
@@ -290,3 +298,88 @@ export function formatYmd( date: Date, timezoneString?: string, gmtOffset?: numb
+/**
+ * Get the initial date range from the URL search parameters.
+ */
+export function getInitialDateRangeFromSearch( search: string ): { start: Date; end: Date } | null {
```
Same as `getDefaultDateRange`, since `getInitialDateRangeFromSearch` is also specific to the `useDateRange` hook, maybe we could just place it in the same file?

myhro replied:
I'm moving it to the hook file. If we ever find another user for it, as it's quite generic, we can move it again.

Initsogar commented on client/dashboard/utils/datetime.ts:
```diff
@@ -290,3 +298,88 @@ export function formatYmd( date: Date, timezoneString?: string, gmtOffset?: numb
+ * Convert a local date range to inclusive epoch-second boundaries (UTC).
+ * Covers the full start and end calendar days.
+ */
+export function buildTimeRangeInSeconds(
```
`buildTimeRangeInSeconds` seems to be specific to logs. Should we consider keeping it co-located with the logs page to maintain clear boundaries between shared utilities and page-specific helpers?

myhro replied:
Yup, I made a confusion about this function. It is indeed logs-specific and shouldn't have been moved here. I'm updating it.

Initsogar commented on client/dashboard/utils/datetime.ts:
```diff
@@ -290,3 +298,88 @@ export function formatYmd( date: Date, timezoneString?: string, gmtOffset?: numb
+ * Creates a full-day time range for activity log filtering, spanning from the start of the
+ * first date to the end of the last date in the specified timezone.
+ */
+export function buildTimeRangeForActivityLog(
```
Is this also intended to be shared with the Activity Log page? It would be nice to give a heads-up to the team working on it so they can reuse it once merged 🙌

myhro replied:
I think it's useful for any Activity Log query using `before`/`after` parameters. I'm moving it to `utils/site-activity-log.ts` to improve its discoverability.

myhro commented:
@Automattic/jetpack-vulcan Hi. Can one of you please review this PR to confirm the changes to the Logs page are fine? Karen was tagged, but apparently will only be back next week.

fushar commented:
> I think someone from Lego would have more knowledge to do an informed PR review than I do.

All right I can do my best to review it tomorrow morning APAC time :)

fushar reviewed (changes requested):
I tested Logs and it seems I found a regression.

The auto-refresh functionality no longer works. If I toggle it on:

<img width="710" height="251" alt="Image" src="https://github.com/user-attachments/assets/9b1844dc-5dc9-4336-a1bf-e22f6ab58f28" />

then the query params should change periodically:

`/v2/sites/:siteSlug/logs/server?from=1757523600&to=1758042000`

Could you also confirm? (and fix)

Initsogar committed:
Fix auto-refresh regression in useDateRange hook

The auto-refresh functionality was broken because the tick function was calling getDefaultDateRange(), which returns a static date range based on "today" rather than a rolling window ending at the current moment.

Restored the original auto-refresh logic that calculates a fresh rolling window every 10 seconds, ensuring URL parameters advance with new log entries.

Initsogar committed:
Merge remote-tracking branch 'origin/trunk' into add/backups-daterangepicker

Initsogar commented:
> I tested Logs and it seems I found a regression.
> The auto-refresh functionality no longer works. If I toggle it on:

@fushar Nice catch. I was able to reproduce the issue and pushed a fix for it (https://github.com/Automattic/wp-calypso/pull/105700/commits/3008d9ba3160ae375f0bbfc0dd10854f542a5f73). Could you take another look?

Looks like we were trying to use the getDefaultDateRange() util for the auto-refresh logic, but that returns a static range ending at “start of today” rather than a rolling window ending now. I restored it to use [the previous logic implemented on the Logs page](https://github.com/Automattic/wp-calypso/blob/2edabcbfa9d82f2554a47c95690f64c388433434/client/dashboard/sites/logs/index.tsx#L123-L131).

StevenDufresne commented:
Over here in Korea, here's what I get for the following:

**Today**

<img width="400" height="847" alt="Screenshot 2025-09-17 at 1 53 33 PM" src="https://github.com/user-attachments/assets/503f0050-1a69-46e9-ba41-4154f077a096" />

Based on what's in the date range picker, i wouldn't expect to see Sept 16 dates in the result set.


**Yesterday**

<img width="400" height="685" alt="Screenshot 2025-09-17 at 1 56 25 PM" src="https://github.com/user-attachments/assets/5f2b70ec-bd2f-4f67-b9e4-f6cf61969c35" />

Same.

fushar reviewed (approved):
I noticed the same thing as @StevenDufresne mentioned. However, I checked that the timestamps in the URL are correct, and the Logs page shows the same thing as production. So, I'm approving it for now; we can fix in a follow-up PR if needed.

My guess is that the issue is related to whether the times are in our local time, UTC, or in the site's timezone.

myhro commented:
Thank you all for the reviews.

> My guess is that the issue is related to whether the times are in our local time, UTC, or in the site's timezone.

I expected that to be covered by the [offset/timezone settings][1], but it looks to be a bit more complicated than that. We can definitely follow-up in another PR.

[1]: https://github.com/Automattic/wp-calypso/pull/105700/files#diff-1a719e72994e1087f94e50650c0fd8fe193906808c872c116418f5601a8f1112R43

myhro commented:
Hi Karen. Thanks for the feedback. 

> However, it looks like the best approach is to be consistent with the site timezone everywhere (including display).

Not sure about Calypso Blue, but apparently that's what we do in Jetpack Cloud:

<img width="716" height="92" alt="image" src="https://github.com/user-attachments/assets/d5236833-8aca-4dca-a9d2-3b5a2a7e935b" />
